### PR TITLE
[Event Hubs Client] Docs Enhancements

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to correct a bug in the samples and to introduce an SDK-level overview, covering each of the libraries.

# Last Upstream Rebase

Thursday, November 19, 2:11pm (EST)

### CODE REVIEW

**Comment by jsquire** [comment]
Once the Overview passes review, I'll reflect it over to the docs repository as well, per discussion.

File: sdk/eventhub/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Azure Event Hubs libraries for .NET
+
+Azure Event Hubs is a highly scalable publish-subscribe service that can ingest millions of events per second and stream them to multiple consumers. This lets you process and analyze the massive amounts of data produced by your connected devices and applications. Once Event Hubs has collected the data, you can retrieve, transform and store it by using any real-time analytics provider or with batching/storage adapters.  If you would like to know more about Azure Event Hubs, you may wish to review: [What is Event Hubs](https://docs.microsoft.com/azure/event-hubs/event-hubs-about)?
```
Comment (christothes) [inline sdk/eventhub/README.md:3]:
```suggestion
Azure Event Hubs is a highly scalable publish-subscribe service that can ingest millions of events per second and stream them to multiple consumers. This lets you process and analyze the massive amounts of data produced by your connected devices and applications. Once Event Hubs has collected the data, you can retrieve, transform, and store it by using any real-time analytics provider or with batching/storage adapters.  If you would like to know more about Azure Event Hubs, you may wish to review: [What is Event Hubs](https://docs.microsoft.com/azure/event-hubs/event-hubs-about).
```

File: sdk/eventhub/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Use the following packages to publish and consume events from Event Hubs:
+
+| Nuget Package | Reference | Samples |
+|--------------------------------------|---------------------------------------------------------------|-------------------------------------------------------------------------------|
+| [Azure.Messaging.EventHubs](https://www.nuget.org/packages/Azure.Messaging.EventHubs)  |  [API Reference for Azure.Messaging.EventHubs](https://docs.microsoft.com/dotnet/api/azure.messaging.eventhubs?view=azure-dotnet)  |  [Samples for Azure.Messaging.EventHubs](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs/samples)  |
+| [Azure.Messaging.EventHubs.Processor](https://www.nuget.org/packages/Azure.Messaging.EventHubs.Processor)  |  [API Reference for Azure.Messaging.EventHubs.Processor](https://docs.microsoft.com/dotnet/api/azure.messaging.eventhubs?view=azure-dotnet)  |  [Samples for Azure.Messaging.EventHubs.Processor](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples)  |
```
Comment (christothes) [inline sdk/eventhub/README.md:18]:
```suggestion
| [Azure.Messaging.EventHubs.Processor](https://www.nuget.org/packages/Azure.Messaging.EventHubs.Processor)  |  [API Reference for Azure.Messaging.EventHubs.Processor](https://docs.microsoft.com/dotnet/api/azure.messaging.eventhubs.processor?view=azure-dotnet)  |  [Samples for Azure.Messaging.EventHubs.Processor](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples)  |
```

**Review by christothes — APPROVED** [review]
Minor tweaks.

File: sdk/eventhub/README.md (hunk 1 above)
Comment (jsquire) [reply sdk/eventhub/README.md:3]: You know... this was largely a copy/paste from the product docs page.  As a fan of the oxford comma, I am ashamed of this oversight.

Comment (jsquire) [reply sdk/eventhub/README.md:3]: I've also reflected this to the package README files.

File: sdk/eventhub/README.md (hunk 2 above)
Comment (jsquire) [reply sdk/eventhub/README.md:18]: I want them to land on the main namespace for this, as that is where `EventProcessorClient` lives;  moving to `.processor` brings you to the event args and other support types.

**Pull request merged**